### ISSUE TITLE
HBO MAX for International regions

### ISSUE THREAD

**Issue by mflorezm** [matthuisman/slyguy.addons#13]
Hi Matt:

Since today, HBO MAX began to work on Latin America Region, so, no more USA IP address needed on my Samsung Tv, nevertheless, the Kodi Addon still tied to the USA API, so, the addon can not be used on Colombia. Is it possible for you add support for other regions? 

Thanks.

**Comment by matthuisman** [comment]
hi @mflorezm 
should be possible.
I would need a test login temporarily so I can test and confirm etc.

if you can help, please change your HBo password to something unique just for it.
Then email me at blog@matthuisman.nz with the username / password

**Comment by rokam** [comment]
To solve this, you should use another Base URL for LATAM: comet-latam.api.hbo.com

**Comment by matthuisman** [comment]
That's only for a few countries. Some use -usa. And maybe others in the future . There is an endpoint that returns the urls to use. So in process of using that.

**Comment by matthuisman** [comment]
@rokam  @mflorezm 
please try v0.2.8 of the add-on.
I sort of went in "blind" without an add-on to test, but hopefully it works.
If not - it will be close to working

**Comment by rokam** [comment]
It's working fine. I'll report any bug.

**Comment by mflorezm** [comment]
Unfortunately, to me 0.2.8 is not working on Kodi 18.9 on librelec 9.2.6 (RPi4) is not working. It is crashing since the openning.

below my log when trying to open the add-on twice:

2021-07-01 08:35:13.132 T:3011727376   ERROR: Misplaced !
2021-07-01 08:35:13.132 T:3011727376   ERROR: Error parsing boolean expression string.endswith(listitem.label,!)
2021-07-01 08:35:14.266 T:2215940976   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.SyntaxError'>
                                            Error Contents: invalid syntax (api.py, line 53)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/slyguy.hbo.max/default.py", line 3, in <module>
                                                from resources.lib.plugin import plugin
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/plugin.py", line 10, in <module>
                                                from .api import API
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/api.py", line 53
                                                return 'https://gateway{userSubdomain}.{domain}.hbo.com{path}'.format(**config['routeKeys'], path=config['endpoints']['getTokens']['path'])
                                                                                                                                           ^
                                            SyntaxError: invalid syntax
                                            -->End of Python script error report<--
2021-07-01 08:35:14.398 T:2309444464   ERROR: GetDirectory - Error getting plugin://slyguy.hbo.max/
2021-07-01 08:35:14.416 T:3011727376   ERROR: CGUIMediaWindow::GetDirectory(plugin://slyguy.hbo.max/) failed
2021-07-01 08:35:14.567 T:3011727376 WARNING: Skin has invalid include: Animation_DialogPopupOpenClose
2021-07-01 08:39:20.530 T:3011727376 WARNING: Previous line repeats 1 times.
2021-07-01 08:39:20.530 T:3011727376   ERROR: Control 50 in window 10025 has been asked to focus, but it can't
2021-07-01 08:39:21.099 T:2600248176   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.SyntaxError'>
                                            Error Contents: invalid syntax (api.py, line 53)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/slyguy.hbo.max/default.py", line 3, in <module>
                                                from resources.lib.plugin import plugin
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/plugin.py", line 10, in <module>
                                                from .api import API
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/api.py", line 53
                                                return 'https://gateway{userSubdomain}.{domain}.hbo.com{path}'.format(**config['routeKeys'], path=config['endpoints']['getTokens']['path'])
                                                                                                                                           ^
                                            SyntaxError: invalid syntax
                                            -->End of Python script error report<--
2021-07-01 08:39:21.222 T:2554106736   ERROR: GetDirectory - Error getting plugin://slyguy.hbo.max/
2021-07-01 08:39:21.250 T:3011727376   ERROR: CGUIMediaWindow::GetDirectory(plugin://slyguy.hbo.max/) failed
2021-07-01 08:39:21.293 T:3011727376 WARNING: Skin has invalid include: Animation_DialogPopupOpenClose
 
Thanks,

**Comment by aranofacundo** [comment]
Hi,
I have the same problem with 0.2.8. In my case I'm using LibreELEC 9.2.7 (Kodi 18.9) with a Raspberry Pi 3B+. I will be using it from Argentina.
```
2021-07-01 07:55:35.954 T:1069998848   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.SyntaxError'>
                                            Error Contents: invalid syntax (api.py, line 53)
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/slyguy.hbo.max/default.py", line 3, in <module>
                                                from resources.lib.plugin import plugin
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/plugin.py", line 10, in <module>
                                                from .api import API
                                              File "/storage/.kodi/addons/slyguy.hbo.max/resources/lib/api.py", line 53
                                                return 'https://gateway{userSubdomain}.{domain}.hbo.com{path}'.format(**config['routeKeys'], path=config['endpoints']['getTokens']['path'])
                                                                                                                                           ^
                                            SyntaxError: invalid syntax
                                            -->End of Python script error report<--
2021-07-01 07:55:36.274 T:1086784256   ERROR: GetDirectory - Error getting plugin://slyguy.hbo.max/
2021-07-01 07:55:36.276 T:1936957504   ERROR: CGUIMediaWindow::GetDirectory(plugin://slyguy.hbo.max/) failed
```
Also, thanks for the plugin.

**Comment by matthuisman** [comment]
oops. forgot to test in Kodi 18 (Python2)
Fixed now with  0.2.9

**Comment by aranofacundo** [comment]
Hi
Sorry for the late reply, now it's working correctly. Also I can send you a Spanish translation of the plugin if you want.
